src/sql/azdata.proposed.d.ts

```diff
@@ -1741,8 +1741,172 @@ declare module 'azdata' {
 		ownerUri: string;
 	}
 
+	export interface SchemaCompareParams {
+		sourceEndpointInfo: SchemaCompareEndpointInfo;
+		targetEndpointInfo: SchemaCompareEndpointInfo;
+		taskExecutionMode: TaskExecutionMode;
+		deploymentOptions: DeploymentOptions;
+	}
+
+	export interface SchemaCompareGenerateScriptParams {
+		operationId: string;
+		targetDatabaseName: string;
+		scriptFilePath: string;
+		taskExecutionMode: TaskExecutionMode;
+	}
+
+	export interface DeploymentOptions {
```

Comment:
Can you add a test that fails when a new option is introduced in DacFx, but not present here?

Reply:
We added a test sqltools side for comparing the sqltoolsservice\DeploymentOptions with dacDeployOptions. the changes between azdata.DeploymentOptions and SqltoolsService\DeploymentOptions should automatically get caught since serialization should fail if they don't match. 

Let me know if you still think we should add another set. We will add a test on sqltools service side since that gets the options with every call.

Reply:
As discussed added test on sqltoolsservice